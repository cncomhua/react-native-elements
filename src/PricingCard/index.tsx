import { withTheme } from '../config';
import { PricingCard, PricingCardProps } from './PricingCard';

export { PricingCard };
export type { PricingCardProps };
/**
 * RNE PricingCard
 *
 * **Preview**
 *
 * ![URL](URL)
 *
 * [**Documentation**](https://reactnativeelements.com/docs/pricingcard)
 */
export default withTheme(PricingCard, 'PricingCard');
