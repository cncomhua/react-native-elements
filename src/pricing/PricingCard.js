import PropTypes from 'prop-types';
import React from 'react';
import { View, StyleSheet, Platform } from 'react-native';
import Text from '../text/Text';
import fonts from '../config/fonts';
import colors from '../config/colors';
import Button from '../buttons/Button';
import Icon from '../icons/Icon';
import normalize from '../helpers/normalizeText';
import elevation from '../config/elevation';
import ViewPropTypes from '../config/ViewPropTypes';

const PricingCard = props => {
  const {
    containerStyle,
    wrapperStyle,
    title,
    price,
    info,
    button,
    color,
    titleFont,
    pricingFont,
    infoFont,
    buttonFont,
    onButtonPress,
    ...attributes
  } = props;
  return (
    <View
      {...attributes}
      style={[styles.container, containerStyle && containerStyle]}
    >
      <View style={[styles.wrapper, wrapperStyle && wrapperStyle]}>
        <Text
          style={[
            styles.pricingTitle,
            { color },
            titleFont && { fontFamily: titleFont },
          ]}
        >
          {title}
        </Text>
        <Text
          style={[
            styles.pricingPrice,
            pricingFont && { fontFamily: pricingFont },
          ]}
        >
          {price}
        </Text>
        {info.map((item, i) => {
          return (
            <Text
              key={i}
              style={[styles.pricingInfo, infoFont && { fontFamily: infoFont }]}
            >
              {item}
            </Text>
          );
        })}
        <Button
          text={button.title}
          buttonStyle={[
            styles.button,
            button.buttonStyle,
            { backgroundColor: color },
          ]}
          onPress={onButtonPress}
          icon={<Icon name={button.icon} size={15} color="white" />}
        />
      </View>
    </View>
  );
};

PricingCard.propTypes = {
  containerStyle: ViewPropTypes.style,
  wrapperStyle: ViewPropTypes.style,
  title: PropTypes.string.isRequired,
  price: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  info: PropTypes.array.isRequired,
  button: PropTypes.object.isRequired,
  color: PropTypes.string.isRequired,
  onButtonPress: PropTypes.any,
  titleFont: PropTypes.string,
  pricingFont: PropTypes.string,
  infoFont: PropTypes.string,
  buttonFont: PropTypes.string,
};

PricingCard.defaultProps = {
  color: colors.primary,
};

const styles = StyleSheet.create({
  container: {
    margin: 15,
    marginBottom: 15,
    backgroundColor: 'white',
    borderColor: colors.grey5,
    borderWidth: 1,
    padding: 15,
    ...Platform.select({
      ios: {
        shadowColor: 'rgba(0,0,0, .2)',
        shadowOffset: { height: 1, width: 0 },
        shadowOpacity: 0.5,
        shadowRadius: 0.5,
      },
      android: {
        ...elevation.one,
      },
      web: {
        ...elevation.one,
      },
    }),
  },
  wrapper: {
    backgroundColor: 'transparent',
  },
  pricingTitle: {
    textAlign: 'center',
    color: colors.primary,
    fontSize: normalize(30),
    ...Platform.select({
      ios: {
        fontWeight: '800',
      },
      android: {
        ...fonts.android.black,
      },
      web: {
        fontWeight: '700',
      },
    }),
  },
  pricingPrice: {
    textAlign: 'center',
    marginTop: 10,
    marginBottom: 10,
    fontSize: normalize(40),
    ...Platform.select({
      ios: {
        fontWeight: '700',
      },
      android: {
        ...fonts.android.bold,
      },
      web: {
        fontWeight: '600',
      },
    }),
  },
  pricingInfo: {
    textAlign: 'center',
    marginTop: 5,
    marginBottom: 5,
    color: colors.grey3,
    ...Platform.select({
      ios: {
        fontWeight: '600',
      },
      android: {
        ...fonts.android.bold,
      },
      web: {
        fontWeight: '600',
      },
    }),
  },
  button: {
    marginTop: 15,
    marginBottom: 10,
  },
});

export default PricingCard;
