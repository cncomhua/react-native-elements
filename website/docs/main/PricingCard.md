---


id: pricingcard


title: PricingCard


slug: /pricingcard

---



import Usage from './usage/PricingCard/PricingCard.md'



Pricing is a convenience component used to display features and pricing tables in a beautiful and engaging way.



## Usage


<Usage />

---


## Props

### PricingCard




## Reference

### PricingCard

