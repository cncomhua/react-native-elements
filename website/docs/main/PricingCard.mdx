---
id: pricingcard

title: PricingCard

slug: /pricingcard
---

import Usage from './usage/PricingCard/PricingCard.mdx';

import Props from './props/PricingCard.mdx';

import { PricingCard } from 'react-native-elements';

import Tabs from '@theme/Tabs';

import TabItem from '@theme/TabItem';

Pricing is a convenience component used to display features and pricing tables in a beautiful and engaging way.

## Usage

<Usage />

## Props

<Props />
