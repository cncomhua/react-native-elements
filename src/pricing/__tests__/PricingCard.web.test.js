import React from 'react';
import { Text, TouchableHighlight, View } from 'react-native';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import parseSamples, {
  assignRef,
  genRefId,
} from 'enzyme-styleguidist-sample-parser';
import options, {
  snapShot,
  buildJsxForGuideMethod,
  ensureCalled,
  onlyEnsureCalled,
  onlySnapshots,
} from '../../../samples';
import PricingCard from '../PricingCard';

const props = {
  'required props: title, price, color, button, & info': {
    component: PricingCard,
    props: {
      title: 'Title',
      price: '$71',
      color: '#071',
      button: {
        title: 'BUTTON TEXT',
        icon: 'pets',
      },
      info: ['info0', 'info1'],
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  'onButtonPress & required props': {
    component: PricingCard,
    props: {
      title: 'Title',
      price: '$71',
      color: '#071',
      button: {
        title: 'BUTTON TEXT',
        icon: 'pets',
      },
      info: ['info0', 'info1'],
      onButtonPress: () => {
        console.log('button pressed');
      },
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  'containerStyle & required props': {
    component: PricingCard,
    props: {
      title: 'Title',
      price: '$71',
      color: '#071',
      button: {
        title: 'BUTTON TEXT',
        icon: 'pets',
      },
      info: ['info0', 'info1'],
      containerStyle: { borderWidth: 5 },
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  'wrapperStyle & required props': {
    component: PricingCard,
    props: {
      title: 'Title',
      price: '$71',
      color: '#071',
      button: {
        title: 'BUTTON TEXT',
        icon: 'pets',
      },
      info: ['info0', 'info1'],
      wrapperStyle: { borderWidth: 5 },
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  'titleFont & required props': {
    component: PricingCard,
    props: {
      title: 'Title',
      price: '$71',
      color: '#071',
      button: {
        title: 'BUTTON TEXT',
        icon: 'pets',
      },
      info: ['info0', 'info1'],
      titleFont: 'Courier New',
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  'pricingFont & required props': {
    component: PricingCard,
    props: {
      title: 'Title',
      price: '$71',
      color: '#071',
      button: {
        title: 'BUTTON TEXT',
        icon: 'pets',
      },
      info: ['info0', 'info1'],
      pricingFont: 'Courier New',
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  'infoFont & required props': {
    component: PricingCard,
    props: {
      title: 'Title',
      price: '$71',
      color: '#071',
      button: {
        title: 'BUTTON TEXT',
        icon: 'pets',
      },
      info: ['info0', 'info1'],
      infoFont: 'Courier New',
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  'buttonFont & required props': {
    component: PricingCard,
    props: {
      title: 'Title',
      price: '$71',
      color: '#071',
      button: {
        title: 'BUTTON TEXT',
        icon: 'pets',
      },
      info: ['info0', 'info1'],
      buttonFont: 'Courier New',
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
};

const samples = {
  props,
};

parseSamples(
  {
    Cards: { sectionComponents: { PricingCard: { samples } } },
  },
  options
);

describe('PricingCard component', () => {
  it('should render without issues', () => {
    const component = shallow(
      <PricingCard
        info={['1 User', 'Basic Support', 'All Core Features']}
        button={{ title: 'GET STARTED', icon: 'flight-takeoff' }}
      />
    );

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render with props', () => {
    const component = shallow(
      <PricingCard
        title="Free"
        price="$0"
        info={['1 User', 'Basic Support', 'All Core Features']}
        button={{ title: 'GET STARTED', icon: 'flight-takeoff' }}
        titleFont="arial"
        infoFont="arial"
        pricingFont="arial"
        buttonFont="arial"
        containerStyle={{ backgroundColor: 'peru' }}
        wrapperStyle={{ backgroundColor: 'peru' }}
      />
    );

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });
});
