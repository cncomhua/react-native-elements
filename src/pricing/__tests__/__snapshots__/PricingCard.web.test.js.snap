// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards: PricingCard: Props Cards: PricingCard: [ prop: buttonFont & required props ] enzyme-shallow: snapshot: Cards: PricingCard: [ prop: buttonFont & required props ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderColor": "#e1e8ee",
      "borderWidth": 1,
      "margin": 15,
      "marginBottom": 15,
      "padding": 15,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
      }
    }
  >
    <TextElement
      style={
        Object {
          "color": "#071",
          "fontSize": 30,
          "fontWeight": "700",
          "textAlign": "center",
        }
      }
    >
      Title
    </TextElement>
    <TextElement
      style={
        Object {
          "fontSize": 40,
          "fontWeight": "600",
          "marginBottom": 10,
          "marginTop": 10,
          "textAlign": "center",
        }
      }
    >
      $71
    </TextElement>
    <TextElement
      key="0"
      style={
        Object {
          "color": "#86939e",
          "fontWeight": "600",
          "marginBottom": 5,
          "marginTop": 5,
          "textAlign": "center",
        }
      }
    >
      info0
    </TextElement>
    <TextElement
      key="1"
      style={
        Object {
          "color": "#86939e",
          "fontWeight": "600",
          "marginBottom": 5,
          "marginTop": 5,
          "textAlign": "center",
        }
      }
    >
      info1
    </TextElement>
    <Button
      TouchableComponent={[Function]}
      ViewComponent={[Function]}
      buttonStyle={
        Array [
          86,
          undefined,
          Object {
            "backgroundColor": "#071",
          },
        ]
      }
      clear={false}
      icon={
        <Icon
          color="white"
          name="pets"
          raised={false}
          reverse={false}
          reverseColor="white"
          size={15}
          underlayColor="white"
        />
      }
      iconRight={false}
      loadingProps={
        Object {
          "color": "white",
          "size": "small",
        }
      }
      onPress={[Function]}
      text="BUTTON TEXT"
    />
  </View>
</View>
`;

exports[`Cards: PricingCard: Props Cards: PricingCard: [ prop: containerStyle & required props ] enzyme-shallow: snapshot: Cards: PricingCard: [ prop: containerStyle & required props ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderColor": "#e1e8ee",
      "borderWidth": 5,
      "margin": 15,
      "marginBottom": 15,
      "padding": 15,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
      }
    }
  >
    <TextElement
      style={
        Object {
          "color": "#071",
          "fontSize": 30,
          "fontWeight": "700",
          "textAlign": "center",
        }
      }
    >
      Title
    </TextElement>
    <TextElement
      style={
        Object {
          "fontSize": 40,
          "fontWeight": "600",
          "marginBottom": 10,
          "marginTop": 10,
          "textAlign": "center",
        }
      }
    >
      $71
    </TextElement>
    <TextElement
      key="0"
      style={
        Object {
          "color": "#86939e",
          "fontWeight": "600",
          "marginBottom": 5,
          "marginTop": 5,
          "textAlign": "center",
        }
      }
    >
      info0
    </TextElement>
    <TextElement
      key="1"
      style={
        Object {
          "color": "#86939e",
          "fontWeight": "600",
          "marginBottom": 5,
          "marginTop": 5,
          "textAlign": "center",
        }
      }
    >
      info1
    </TextElement>
    <Button
      TouchableComponent={[Function]}
      ViewComponent={[Function]}
      buttonStyle={
        Array [
          86,
          undefined,
          Object {
            "backgroundColor": "#071",
          },
        ]
      }
      clear={false}
      icon={
        <Icon
          color="white"
          name="pets"
          raised={false}
          reverse={false}
          reverseColor="white"
          size={15}
          underlayColor="white"
        />
      }
      iconRight={false}
      loadingProps={
        Object {
          "color": "white",
          "size": "small",
        }
      }
      onPress={[Function]}
      text="BUTTON TEXT"
    />
  </View>
</View>
`;

exports[`Cards: PricingCard: Props Cards: PricingCard: [ prop: infoFont & required props ] enzyme-shallow: snapshot: Cards: PricingCard: [ prop: infoFont & required props ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderColor": "#e1e8ee",
      "borderWidth": 1,
      "margin": 15,
      "marginBottom": 15,
      "padding": 15,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
      }
    }
  >
    <TextElement
      style={
        Object {
          "color": "#071",
          "fontSize": 30,
          "fontWeight": "700",
          "textAlign": "center",
        }
      }
    >
      Title
    </TextElement>
    <TextElement
      style={
        Object {
          "fontSize": 40,
          "fontWeight": "600",
          "marginBottom": 10,
          "marginTop": 10,
          "textAlign": "center",
        }
      }
    >
      $71
    </TextElement>
    <TextElement
      key="0"
      style={
        Object {
          "color": "#86939e",
          "fontFamily": "Courier New",
          "fontWeight": "600",
          "marginBottom": 5,
          "marginTop": 5,
          "textAlign": "center",
        }
      }
    >
      info0
    </TextElement>
    <TextElement
      key="1"
      style={
        Object {
          "color": "#86939e",
          "fontFamily": "Courier New",
          "fontWeight": "600",
          "marginBottom": 5,
          "marginTop": 5,
          "textAlign": "center",
        }
      }
    >
      info1
    </TextElement>
    <Button
      TouchableComponent={[Function]}
      ViewComponent={[Function]}
      buttonStyle={
        Array [
          86,
          undefined,
          Object {
            "backgroundColor": "#071",
          },
        ]
      }
      clear={false}
      icon={
        <Icon
          color="white"
          name="pets"
          raised={false}
          reverse={false}
          reverseColor="white"
          size={15}
          underlayColor="white"
        />
      }
      iconRight={false}
      loadingProps={
        Object {
          "color": "white",
          "size": "small",
        }
      }
      onPress={[Function]}
      text="BUTTON TEXT"
    />
  </View>
</View>
`;

exports[`Cards: PricingCard: Props Cards: PricingCard: [ prop: onButtonPress & required props ] enzyme-shallow: snapshot: Cards: PricingCard: [ prop: onButtonPress & required props ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderColor": "#e1e8ee",
      "borderWidth": 1,
      "margin": 15,
      "marginBottom": 15,
      "padding": 15,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
      }
    }
  >
    <TextElement
      style={
        Object {
          "color": "#071",
          "fontSize": 30,
          "fontWeight": "700",
          "textAlign": "center",
        }
      }
    >
      Title
    </TextElement>
    <TextElement
      style={
        Object {
          "fontSize": 40,
          "fontWeight": "600",
          "marginBottom": 10,
          "marginTop": 10,
          "textAlign": "center",
        }
      }
    >
      $71
    </TextElement>
    <TextElement
      key="0"
      style={
        Object {
          "color": "#86939e",
          "fontWeight": "600",
          "marginBottom": 5,
          "marginTop": 5,
          "textAlign": "center",
        }
      }
    >
      info0
    </TextElement>
    <TextElement
      key="1"
      style={
        Object {
          "color": "#86939e",
          "fontWeight": "600",
          "marginBottom": 5,
          "marginTop": 5,
          "textAlign": "center",
        }
      }
    >
      info1
    </TextElement>
    <Button
      TouchableComponent={[Function]}
      ViewComponent={[Function]}
      buttonStyle={
        Array [
          86,
          undefined,
          Object {
            "backgroundColor": "#071",
          },
        ]
      }
      clear={false}
      icon={
        <Icon
          color="white"
          name="pets"
          raised={false}
          reverse={false}
          reverseColor="white"
          size={15}
          underlayColor="white"
        />
      }
      iconRight={false}
      loadingProps={
        Object {
          "color": "white",
          "size": "small",
        }
      }
      onPress={[Function]}
      text="BUTTON TEXT"
    />
  </View>
</View>
`;

exports[`Cards: PricingCard: Props Cards: PricingCard: [ prop: pricingFont & required props ] enzyme-shallow: snapshot: Cards: PricingCard: [ prop: pricingFont & required props ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderColor": "#e1e8ee",
      "borderWidth": 1,
      "margin": 15,
      "marginBottom": 15,
      "padding": 15,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
      }
    }
  >
    <TextElement
      style={
        Object {
          "color": "#071",
          "fontSize": 30,
          "fontWeight": "700",
          "textAlign": "center",
        }
      }
    >
      Title
    </TextElement>
    <TextElement
      style={
        Object {
          "fontFamily": "Courier New",
          "fontSize": 40,
          "fontWeight": "600",
          "marginBottom": 10,
          "marginTop": 10,
          "textAlign": "center",
        }
      }
    >
      $71
    </TextElement>
    <TextElement
      key="0"
      style={
        Object {
          "color": "#86939e",
          "fontWeight": "600",
          "marginBottom": 5,
          "marginTop": 5,
          "textAlign": "center",
        }
      }
    >
      info0
    </TextElement>
    <TextElement
      key="1"
      style={
        Object {
          "color": "#86939e",
          "fontWeight": "600",
          "marginBottom": 5,
          "marginTop": 5,
          "textAlign": "center",
        }
      }
    >
      info1
    </TextElement>
    <Button
      TouchableComponent={[Function]}
      ViewComponent={[Function]}
      buttonStyle={
        Array [
          86,
          undefined,
          Object {
            "backgroundColor": "#071",
          },
        ]
      }
      clear={false}
      icon={
        <Icon
          color="white"
          name="pets"
          raised={false}
          reverse={false}
          reverseColor="white"
          size={15}
          underlayColor="white"
        />
      }
      iconRight={false}
      loadingProps={
        Object {
          "color": "white",
          "size": "small",
        }
      }
      onPress={[Function]}
      text="BUTTON TEXT"
    />
  </View>
</View>
`;

exports[`Cards: PricingCard: Props Cards: PricingCard: [ prop: required props: title, price, color, button, & info ] enzyme-shallow: snapshot: Cards: PricingCard: [ prop: required props: title, price, color, button, & info ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderColor": "#e1e8ee",
      "borderWidth": 1,
      "margin": 15,
      "marginBottom": 15,
      "padding": 15,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
      }
    }
  >
    <TextElement
      style={
        Object {
          "color": "#071",
          "fontSize": 30,
          "fontWeight": "700",
          "textAlign": "center",
        }
      }
    >
      Title
    </TextElement>
    <TextElement
      style={
        Object {
          "fontSize": 40,
          "fontWeight": "600",
          "marginBottom": 10,
          "marginTop": 10,
          "textAlign": "center",
        }
      }
    >
      $71
    </TextElement>
    <TextElement
      key="0"
      style={
        Object {
          "color": "#86939e",
          "fontWeight": "600",
          "marginBottom": 5,
          "marginTop": 5,
          "textAlign": "center",
        }
      }
    >
      info0
    </TextElement>
    <TextElement
      key="1"
      style={
        Object {
          "color": "#86939e",
          "fontWeight": "600",
          "marginBottom": 5,
          "marginTop": 5,
          "textAlign": "center",
        }
      }
    >
      info1
    </TextElement>
    <Button
      TouchableComponent={[Function]}
      ViewComponent={[Function]}
      buttonStyle={
        Array [
          86,
          undefined,
          Object {
            "backgroundColor": "#071",
          },
        ]
      }
      clear={false}
      icon={
        <Icon
          color="white"
          name="pets"
          raised={false}
          reverse={false}
          reverseColor="white"
          size={15}
          underlayColor="white"
        />
      }
      iconRight={false}
      loadingProps={
        Object {
          "color": "white",
          "size": "small",
        }
      }
      onPress={[Function]}
      text="BUTTON TEXT"
    />
  </View>
</View>
`;

exports[`Cards: PricingCard: Props Cards: PricingCard: [ prop: titleFont & required props ] enzyme-shallow: snapshot: Cards: PricingCard: [ prop: titleFont & required props ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderColor": "#e1e8ee",
      "borderWidth": 1,
      "margin": 15,
      "marginBottom": 15,
      "padding": 15,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
      }
    }
  >
    <TextElement
      style={
        Object {
          "color": "#071",
          "fontFamily": "Courier New",
          "fontSize": 30,
          "fontWeight": "700",
          "textAlign": "center",
        }
      }
    >
      Title
    </TextElement>
    <TextElement
      style={
        Object {
          "fontSize": 40,
          "fontWeight": "600",
          "marginBottom": 10,
          "marginTop": 10,
          "textAlign": "center",
        }
      }
    >
      $71
    </TextElement>
    <TextElement
      key="0"
      style={
        Object {
          "color": "#86939e",
          "fontWeight": "600",
          "marginBottom": 5,
          "marginTop": 5,
          "textAlign": "center",
        }
      }
    >
      info0
    </TextElement>
    <TextElement
      key="1"
      style={
        Object {
          "color": "#86939e",
          "fontWeight": "600",
          "marginBottom": 5,
          "marginTop": 5,
          "textAlign": "center",
        }
      }
    >
      info1
    </TextElement>
    <Button
      TouchableComponent={[Function]}
      ViewComponent={[Function]}
      buttonStyle={
        Array [
          86,
          undefined,
          Object {
            "backgroundColor": "#071",
          },
        ]
      }
      clear={false}
      icon={
        <Icon
          color="white"
          name="pets"
          raised={false}
          reverse={false}
          reverseColor="white"
          size={15}
          underlayColor="white"
        />
      }
      iconRight={false}
      loadingProps={
        Object {
          "color": "white",
          "size": "small",
        }
      }
      onPress={[Function]}
      text="BUTTON TEXT"
    />
  </View>
</View>
`;

exports[`Cards: PricingCard: Props Cards: PricingCard: [ prop: wrapperStyle & required props ] enzyme-shallow: snapshot: Cards: PricingCard: [ prop: wrapperStyle & required props ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderColor": "#e1e8ee",
      "borderWidth": 1,
      "margin": 15,
      "marginBottom": 15,
      "padding": 15,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "borderWidth": 5,
      }
    }
  >
    <TextElement
      style={
        Object {
          "color": "#071",
          "fontSize": 30,
          "fontWeight": "700",
          "textAlign": "center",
        }
      }
    >
      Title
    </TextElement>
    <TextElement
      style={
        Object {
          "fontSize": 40,
          "fontWeight": "600",
          "marginBottom": 10,
          "marginTop": 10,
          "textAlign": "center",
        }
      }
    >
      $71
    </TextElement>
    <TextElement
      key="0"
      style={
        Object {
          "color": "#86939e",
          "fontWeight": "600",
          "marginBottom": 5,
          "marginTop": 5,
          "textAlign": "center",
        }
      }
    >
      info0
    </TextElement>
    <TextElement
      key="1"
      style={
        Object {
          "color": "#86939e",
          "fontWeight": "600",
          "marginBottom": 5,
          "marginTop": 5,
          "textAlign": "center",
        }
      }
    >
      info1
    </TextElement>
    <Button
      TouchableComponent={[Function]}
      ViewComponent={[Function]}
      buttonStyle={
        Array [
          86,
          undefined,
          Object {
            "backgroundColor": "#071",
          },
        ]
      }
      clear={false}
      icon={
        <Icon
          color="white"
          name="pets"
          raised={false}
          reverse={false}
          reverseColor="white"
          size={15}
          underlayColor="white"
        />
      }
      iconRight={false}
      loadingProps={
        Object {
          "color": "white",
          "size": "small",
        }
      }
      onPress={[Function]}
      text="BUTTON TEXT"
    />
  </View>
</View>
`;

exports[`PricingCard component should render with props 1`] = `
<View
  style={
    Object {
      "backgroundColor": "peru",
      "borderColor": "#e1e8ee",
      "borderWidth": 1,
      "margin": 15,
      "marginBottom": 15,
      "padding": 15,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "peru",
      }
    }
  >
    <TextElement
      style={
        Object {
          "color": "#2089dc",
          "fontFamily": "arial",
          "fontSize": 30,
          "fontWeight": "700",
          "textAlign": "center",
        }
      }
    >
      Free
    </TextElement>
    <TextElement
      style={
        Object {
          "fontFamily": "arial",
          "fontSize": 40,
          "fontWeight": "600",
          "marginBottom": 10,
          "marginTop": 10,
          "textAlign": "center",
        }
      }
    >
      $0
    </TextElement>
    <TextElement
      key="0"
      style={
        Object {
          "color": "#86939e",
          "fontFamily": "arial",
          "fontWeight": "600",
          "marginBottom": 5,
          "marginTop": 5,
          "textAlign": "center",
        }
      }
    >
      1 User
    </TextElement>
    <TextElement
      key="1"
      style={
        Object {
          "color": "#86939e",
          "fontFamily": "arial",
          "fontWeight": "600",
          "marginBottom": 5,
          "marginTop": 5,
          "textAlign": "center",
        }
      }
    >
      Basic Support
    </TextElement>
    <TextElement
      key="2"
      style={
        Object {
          "color": "#86939e",
          "fontFamily": "arial",
          "fontWeight": "600",
          "marginBottom": 5,
          "marginTop": 5,
          "textAlign": "center",
        }
      }
    >
      All Core Features
    </TextElement>
    <Button
      TouchableComponent={[Function]}
      ViewComponent={[Function]}
      buttonStyle={
        Array [
          86,
          undefined,
          Object {
            "backgroundColor": "#2089dc",
          },
        ]
      }
      clear={false}
      icon={
        <Icon
          color="white"
          name="flight-takeoff"
          raised={false}
          reverse={false}
          reverseColor="white"
          size={15}
          underlayColor="white"
        />
      }
      iconRight={false}
      loadingProps={
        Object {
          "color": "white",
          "size": "small",
        }
      }
      onPress={[Function]}
      text="GET STARTED"
    />
  </View>
</View>
`;

exports[`PricingCard component should render without issues 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderColor": "#e1e8ee",
      "borderWidth": 1,
      "margin": 15,
      "marginBottom": 15,
      "padding": 15,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
      }
    }
  >
    <TextElement
      style={
        Object {
          "color": "#2089dc",
          "fontSize": 30,
          "fontWeight": "700",
          "textAlign": "center",
        }
      }
    />
    <TextElement
      style={
        Object {
          "fontSize": 40,
          "fontWeight": "600",
          "marginBottom": 10,
          "marginTop": 10,
          "textAlign": "center",
        }
      }
    />
    <TextElement
      key="0"
      style={
        Object {
          "color": "#86939e",
          "fontWeight": "600",
          "marginBottom": 5,
          "marginTop": 5,
          "textAlign": "center",
        }
      }
    >
      1 User
    </TextElement>
    <TextElement
      key="1"
      style={
        Object {
          "color": "#86939e",
          "fontWeight": "600",
          "marginBottom": 5,
          "marginTop": 5,
          "textAlign": "center",
        }
      }
    >
      Basic Support
    </TextElement>
    <TextElement
      key="2"
      style={
        Object {
          "color": "#86939e",
          "fontWeight": "600",
          "marginBottom": 5,
          "marginTop": 5,
          "textAlign": "center",
        }
      }
    >
      All Core Features
    </TextElement>
    <Button
      TouchableComponent={[Function]}
      ViewComponent={[Function]}
      buttonStyle={
        Array [
          86,
          undefined,
          Object {
            "backgroundColor": "#2089dc",
          },
        ]
      }
      clear={false}
      icon={
        <Icon
          color="white"
          name="flight-takeoff"
          raised={false}
          reverse={false}
          reverseColor="white"
          size={15}
          underlayColor="white"
        />
      }
      iconRight={false}
      loadingProps={
        Object {
          "color": "white",
          "size": "small",
        }
      }
      onPress={[Function]}
      text="GET STARTED"
    />
  </View>
</View>
`;
